haystack/document_store/faiss.py

```diff
@@ -455,17 +473,38 @@ def load(
 
         :param faiss_file_path: Stored FAISS index file. Can be created via calling `save()`
         :param sql_url: Connection string to the SQL database that contains your docs and metadata.
+            Overrides the value defined in the `faiss_init_params_path` file, if present
         :param index: Index name to load the FAISS index as. It must match the index name used for
-                      when creating the FAISS index.
-        :return:
-        """
-        """
+                      when creating the FAISS index. Overrides the value defined in the 
+                      `faiss_init_params_path` file, if present
+        :param faiss_init_params_path: Stored FAISS init parameters. Can be created via calling `save()`
+        :return: the DocumentStore
         """
+        if not faiss_init_params_path:
+            faiss_file_path = Path(faiss_file_path)
+            faiss_init_params_path = faiss_file_path.with_suffix(".json")
+
+        with open(faiss_init_params_path, 'r') as ipp:
+            init_params = json.load(ipp)
+
         faiss_index = faiss.read_index(str(faiss_file_path))
-        return cls(
-            faiss_index=faiss_index,
-            sql_url=sql_url,
-            vector_dim=faiss_index.d,
-            index=index,
-        )
 
+        # Add other init params to override the ones defined in the init params file
+        init_params["faiss_index"] = faiss_index
+        init_params["vector_dim"]=faiss_index.d
+        
+        # NOTE: The following block could be removed if we don't care about backwards API compatibility
+        # (along with the sql_url and index parameters, which now can become confusing)
```

Comment:
I think it's reasonable to go for the breaking change here and remove both args. 

For those cases of adding breaking changes: please add the label "breaking change" to the PR and provide a small code example for the "old" and "new" version (we will use this in the release notes later on)